controllers/istiorevision/istiorevision_controller.go

```diff
@@ -213,6 +210,14 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 	// The handler triggers the reconciliation of the referenced IstioRevision CR so that its InUse condition is updated.
 	podHandler := wrapEventHandler(logger, handler.EnqueueRequestsFromMapFunc(r.mapPodToReconcileRequest))
 
+	// revisionTagHandler handles Istios that reference an IstioRevision CR as ActiveRevision
+	// The handler triggers the reconciliation of the active IstioRevision CR so that its InUse condition is updated.
+	istioHandler := wrapEventHandler(logger, handler.EnqueueRequestsFromMapFunc(r.mapIstioToReconcileRequest))
```

Comment:
This fixes a bug I found while testing. It can only occur if no workload or namespace is labelled with the revision tag and/or there are no pod events coming in. The scenario is as follows:

You have a RevisionBased Istio I, a Revision R and a tag T. T references I, and therefore R, R's InUse condition is `true`. Now when you update the version in I, a new revision R* is created, and the active revision of I is updated. This creates a reconcile of T, which is now pointing to R*, which in turn leads to a reconciliation of R*, to mark it InUse=true. Without this watch, R is never reconciled again, which means its InUse condition never flips to `false` - until a Pod event causes its reconciliation.

Reply:
The current approach may also be fragile, since it's possible that it triggers the revision reconciliation before the tag's `status.istioRevision` is updated. I believe there is a chance for a race condition here.